### ISSUE TITLE
Katalina 214U Does Not Display Initial Active Hitbox

### ISSUE THREAD

**Issue by sente-descin** [agersant/gbvsr-frame-meter#1]
Katalina's 214U has a 2 frame active hitbox that extends before she transitions to her main active hitbox. The frame meter does not display the initial active frames even when she connects with the opponent.

**Comment by agersant** [comment]
I started looking into this, it turns out this is the same root cause as the missing active frames on Ferry 6S.

Some notes for when I pick this back up:

The missing active frames belong to projectiles (Beppo / Katalina's summoned thing). They are escaping the current projectile detection because they don't have the `active_frames` flag set. We can't simply change this as there are other entities _with hitboxes_ (??) spawned after a fireball connects, and this flag is how we avoid displaying them.

I have been looking into other ways to distinguish desirable projectiles from noise and so far nothing has been working for all abilities at once:
- Relying on the `is_active()` function from the game (signature `40 53 48 83 EC 20 F6 81 A8 07 00 00 01 48 8B D9 ?? ?? F7 81 A8 01 00 00 00 00`) removes the ghost projectiles. This makes us miss projectiles that hit on frame 1 as they immediately become inactive.
- There's a different hitbox count following the pointer at `0x60` in entity structs, and then at offset `0x2F`. This is set for the problematic projectiles but not basic ones nor ghost projectiles. Seems related to having an animated sprite, and that's also where hitbox/hurtbox data lives for characters.
- Basic projectiles (not Beppo / Katalina 2S+U / unwanted ghost projectiles) have `20` as the first byte of their flags_1 field.

Best attempt so far:
```
for (size_t i = 0; i < ASW::Engine::NUM_ENTITIES; i++)
{
	ASW::Entity *entity = battle->engine->entities[i];
	if (!entity || entity == character || entity->parent_character != character)
	{
		continue;
	}
	const bool is_ghost = ((entity->flags_1 & 0x02) == 0) && !(entity->frame_info && entity->frame_info->num_hitboxes > 0);
	if (entity->num_hitboxes <= 0 || is_ghost)
	{
		continue;
	}
	if (entity->recovery && !entity->attack_hit_connecting)
	{
		continue;
	}
	return CharacterState::PROJECTILE;
}
```

Problems:
1. Very ugly
2. Zeta projectile overstays after hitting as it relies on `active_frames` or `is_active()` to stop acting (not `recovery`).
3. Beppo overstays even without hitting and I can't see any way to tell when its active period ends

**Comment by agersant** [comment]
Fixed in 0a27c3fb18c8cc0f60d0368d3b3b6e18f0f9a2de 🥳
